### PR TITLE
Enable data ready fix

### PR DESCRIPTION
**Pull request by caila-marashaj**
This pr adds hardware support for the data ready pins on the pipette main boards, but does not check that pin's value or incorporate it into the pressure sensor reads. That will be done in a follow-up pr.

### CODE REVIEW

**Review by sfoster1 — APPROVED** [review]
I'm good with this as long as you smoke test it.

File: pipettes/firmware/hardware_config.h [hunk 1]
```diff
@@ -10,7 +10,8 @@ typedef struct {
 typedef enum {
     pipette_hardware_device_LED_drive,
     pipette_hardware_device_sync_in,
-    pipette_hardware_device_sync_out
+    pipette_hardware_device_sync_out,
```
Comment (Laura-Danielle) [inline pipettes/firmware/hardware_config.h:13]: I think this is not a needed change.

File: pipettes/firmware/hardware_config.c [hunk 2]
```diff
@@ -68,7 +68,6 @@ static PipetteHardwarePin get_gpio_lt(PipetteHardwareDevice device) {
     }
 }
 
-
```
Comment (Laura-Danielle) [inline pipettes/firmware/hardware_config.c:71]: same here

**Review by Laura-Danielle — APPROVED** [review]
Two changes that you can get rid of b/c they're unused, but otherwise looks good to merge.

File: pipettes/firmware/hardware_config.c [hunk 3]
```diff
@@ -61,6 +64,10 @@ static PipetteHardwarePin get_gpio_lt(PipetteHardwareDevice device) {
             pinout.port = GPIOB;
             pinout.pin = GPIO_PIN_4;
             return pinout;
+        case pipette_hardware_device_data_ready:
```
Comment (Laura-Danielle) [inline pipettes/firmware/hardware_config.c:67]: This is the wrong function. It should be in `get_gpio_lt`

File: pipettes/firmware/hardware_config.h (hunk 1 above)
Comment (Laura-Danielle) [reply pipettes/firmware/hardware_config.h:13]: Resolving: It is being used, but it wasn't implemented in `hardware_config.c`

**Review by Laura-Danielle — CHANGES_REQUESTED** [review]

**Review by Laura-Danielle — APPROVED** [review]

**Pull request merged**